symfony/plugins/orangehrmTimePlugin/Api/CopyProjectActivityAPI.php

```diff
@@ -0,0 +1,189 @@
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function create(): EndpointResult
+    {
+        list($toProjectId) = $this->getUrlAttributes();
+        $ids = $this->getRequestParams()->getArray(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_ACTIVITY_IDS);
+        $this->getProjectService()->getProjectActivityDao()->saveCopyActivity($toProjectId, $ids);
+        return new EndpointResourceResult(ArrayModel::class, $ids);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForCreate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            $this->getValidationDecorator()->requiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_ACTIVITY_IDS,
+                    new Rule(Rules::ARRAY_TYPE),
+                    new Rule(
+                        Rules::EACH,
+                        [new Rules\Composite\AllOf(new Rule(Rules::POSITIVE))]
+                    )
+                ),
+            ),
+            ...$this->getCommonBodyValidationRules(),
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function delete(): EndpointResult
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForDelete(): ParamRuleCollection
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResult
+    {
+        list($toProjectId, $fromProjectId) = $this->getUrlAttributes();
+        $baseProject = $this->getProjectService()->getProjectDao()->getProjectById($fromProjectId);
+        $this->throwRecordNotFoundExceptionIfNotExist($baseProject, Project::class);
+        $targetProject = $this->getProjectService()->getProjectDao()->getProjectById($toProjectId);
+        $this->throwRecordNotFoundExceptionIfNotExist($targetProject, Project::class);
+
+        $copyActivityField = new CopyActivityField();
+        $copyActivityField->setFromProjectId($fromProjectId);
+        $copyActivityField->setToProjectId($toProjectId);
+
+        return new EndpointResourceResult(CopyActivityModel::class, $copyActivityField);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            $this->getValidationDecorator()->requiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_FROM_PROJECT_ID,
+                    new Rule(Rules::POSITIVE),
+                )
+            ),
+            ...$this->getCommonBodyValidationRules(),
+        );
+    }
+
+    /**
+     * @return ParamRule[]
+     */
+    private function getCommonBodyValidationRules(): array
```

Comment:
method name is wrong. `PARAMETER_TO_PROJECT_ID` not a body parameter

Reply:
done